### PR TITLE
Split widget API types into a separate file

### PR DESCRIPTION
**Pull request by elainefigma**
This way, we can add a process to sync widget-docs and widget-typings. There are no major changes. I moved around type definitions and made minor tweaks to match the widget-docs.

You can look at the commit history to review this PR - select commits after b5b1f2f Move widgetAPI to separate file.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -24,626 +24,6 @@ declare global {
 
   ///////////////////////////////////////////////////////////////////////////////
   // Widget
-
-  interface WidgetAPI {
-    register(component: FunctionalWidget<any>): void
-    h(...args: any[]): FigmaDeclarativeNode
```
Comment (ymichael) [inline index.d.ts:30]: was this line removed intentionally?

File: index.d.ts [hunk 2]
```diff
@@ -24,626 +24,6 @@ declare global {
 
   ///////////////////////////////////////////////////////////////////////////////
   // Widget
```
Comment (ymichael) [inline index.d.ts:26]: lets remove these comments now that the body is gone

File: index.d.ts [hunk 3]
```diff
@@ -24,626 +24,6 @@ declare global {
...
-  }
-
-  interface WidgetAttachedStickablesChangedEvent {
-    stuckNodeIds: string[]
-    unstuckNodeIds: string[]
-  }
-
-  type SyncedMap<T = any> = {
-    /** @deprecated use size instead */
-    readonly length: number
```
Comment (ymichael) [inline index.d.ts:87]: lets keep this around too so we remain backwards compatible with the existing typings

if we remove this, code that uses these will start to fail typecheck when they update typing versions

File: index.d.ts [hunk 4]
```diff
@@ -24,626 +24,6 @@ declare global {
...
-    | 'link'
-
-  type HexCode = string
-  interface PropertyMenuItem {
-    tooltip: string
-    propertyName: string
-    itemType: PropertyMenuItemType
-  }
-
-  interface WidgetPropertyMenuActionItem extends PropertyMenuItem {
```
Comment (ymichael) [inline index.d.ts:146]: is it possible to keep the `extends PropertyMenuItem` pattern here?

File: index.d.ts [hunk 5]
```diff
@@ -24,626 +24,6 @@ declare global {
...
-      type: 'inner-shadow'
-      color: HexCode | Color
-      offset: Vector
-      blur: number
-      blendMode?: BlendMode
-      spread?: number
-      visible?: boolean
-    }
-
-    export type ShadowEffect = DropShadowEffect | InnerShadowEffect
```
Comment (ymichael) [inline index.d.ts:372]: I think this is fine if it is too hard to add back but something we want to be mindful of - we're remove this type alias so anyone who uses the `ShadowEffect` type alias directly will now need to update their code when they update typings.

**Review by ymichael — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (elainefigma) [reply index.d.ts:30]: Yes! As far as I can tell, there is no implementation for a 'h' hook in the widget API: https://github.com/figma/figma/blob/master/web/js/figma_app/plugin/jsvm_figma_api.ts

File: index.d.ts (hunk 5 above)
Comment (elainefigma) [reply index.d.ts:372]: I can add it back.

**Review by ymichael — APPROVED** [review]
thanks!

**Pull request merged**